// @indirecttek/essentials-engine/dist/types.ts
export interface Service {
  name: string;
  description: string;
}

export interface HeroSection {
  headline: string;
  subheadline: string;
  imageUrl: string;
  imageAlt: string;
  callToActionLabel: string;
}

export interface Theme {
  primary: string;
  secondary: string;
  accent: string;
  background: string;
  foreground: string;
}

export interface ContactInfo {
  phone: string;
  email: string;
  address: string;
}

export interface AnalyticsConfig {
  enableTracking: boolean;
  mixpanelToken?: string;
}

export interface SEOConfig {
  title: string;
  description: string;
}

export interface ImageSearchHints {
  hero: string;
  services: string[];
}

export interface SocialLinks {
  facebook?: string;
  instagram?: string;
  twitter?: string;
  linkedin?: string;
  youtube?: string;
  tiktok?: string;
}

export interface LayoutOptions {
  heroLayout?: "image-left" | "image-right" | "full-width";
  stickyNav?: boolean;
}

export interface SiteConfig {
  businessName: string;
  theme: Theme;
  contactInfo: ContactInfo;
  heroSection: HeroSection;
  services: Service[];
  analytics: AnalyticsConfig;
  seo: SEOConfig;
  imageSearchHints?: ImageSearchHints;
  socialLinks?: SocialLinks;
  layoutOptions?: LayoutOptions;
}
